feat: update password component to use uniformed component file

zzggo opened the pull request:
Closes #861

## Related Issue

Closes #861 

<!-- If this PR addresses an issue, link it here  -->

## Summary of Changes

<!-- Provide a concise description of the changes made in this PR.
What functionality was added, updated, or fixed? -->

## Need Regression Testing

Check password input behaviour in login register unlock and change password.

- [x] Yes
- [ ] No

## Risk Assessment

<!-- Assess the risk level of this PR:
- Low: Minimal impact, straightforward changes.
- Medium: Potential for some edge cases or indirect effects.
- High: Could affect critical functionality or many users.
-->

- [ ] Low
- [x] Medium
- [ ] High

## Additional Notes

<!-- Add any other context or comments about the PR here -->

## Screenshots (if applicable)

<!-- Attach any screenshots that help explain your changes -->

zzggo committed:
feat: update password component to use uniformed component file

Closes #861

zzggo committed:
Merge branch 'dev' into 861-feature-consolidate-password-input

Closes #861

tombeckenham commented on src/ui/FRWComponent/LandingPages/PasswordComponents.tsx:
```diff
@@ -87,25 +87,44 @@ interface PasswordInputProps {
+  endAdornment?: React.ReactNode;
 }
 
 export const PasswordInput = ({
```
This should be in FRWComponent rather than in the LandingPages folder, right?

tombeckenham reviewed (changes requested):
I haven't had a good look yet, but please start by moving the password component into it's own file in FRWComponent

zzggo committed:
fixed: move the PasswordComponent under FRWComponent dir

Closes #861

tombeckenham reviewed (changes requested):
It's a good start, but I think we can do better than this. I'd like to see the styles consolodated. e.g. classes.inputBox. I'd like to see the placeholder text centralised, to include the strength when setting a new password, and to handle confimation. That logic is duplicated

zzggo committed:
Merge branch 'dev' into 861-feature-consolidate-password-input

# Conflicts:
#	src/ui/views/RetrievePK/index.tsx
#	src/ui/views/Unlock/index.tsx

Closes #861

zzggo committed:
Feat: added passwordform component, remove duplicated styles used in password input

Closes #861

zzggo committed:
Merge branch 'dev' into 861-feature-consolidate-password-input

Closes #861

zzggo committed:
Fixed: update component for validation check

Closes #861

zzggo commented on src/ui/FRWComponent/PasswordForm.tsx:
```diff
@@ -0,0 +1,191 @@
+          sx={{
+            width: '100%',
+            pb: requireMatch && index === matchFields[0] ? '30px' : '0px',
+            marginTop: index > 0 ? (fields[index - 1].value ? '0px' : '24px') : '0px',
```
Theres already a check for the condition index > 0 ?

zzggo committed:
fixed: add null check on length

Closes #861

zzggo committed:
fixed: null checks on password comparison

Closes #861

zzggo committed:
Merge branch 'dev' into 861-feature-consolidate-password-input

Closes #861

Commit:
Merge branch 'dev' into 861-feature-consolidate-password-input

# Conflicts:
#	src/ui/FRWComponent/LandingPages/PasswordComponents.tsx
#	src/ui/components/LandingPages/PasswordComponents.tsx
#	src/ui/components/LandingPages/SetPassword.tsx
#	src/ui/components/PasswordComponents.tsx
#	src/ui/components/PasswordForm.tsx
#	src/ui/components/SettingsPassword.tsx
#	src/ui/views/Forgot/Recover/RecoverPage.tsx
#	src/ui/views/Setting/change-password/index.tsx
#	src/ui/views/Setting/recoveryphase/Recoveryphrasepassword.tsx
#	src/ui/views/Unlock/index.tsx
#	src/ui/views/Welcome/AccountImport/Google/DecryptWallet.tsx
#	src/ui/views/Welcome/AccountImport/Google/GoogleRecoverPassword.tsx

Closes #861

Commit:
fixed: unresolved password box

Closes #861

Commit:
fixed: remove mui style

Closes #861

Commit:
fixed: removed unuse margin style

Closes #861

Commit:
Merge branch 'dev' into 861-feature-consolidate-password-input

# Conflicts:
#	src/ui/views/Unlock/index.tsx

Closes #861

Commit:
fixed: remove duplicate

Closes #861

Commit:
fixed: component structure and file linting

Closes #861

Commit:
fixed: update password component structure

Closes #861

tombeckenham commented on src/ui/components/password/PasswordInput.tsx:
```diff
@@ -0,0 +1,122 @@
+export const PasswordInput = ({
+  value,
+  onChange,
+  isVisible,
```
Maybe change this to showPassword, and setShowPassword

tombeckenham commented on src/ui/components/password/PasswordInput.tsx:
```diff
@@ -0,0 +1,122 @@
+  readOnly = false,
+  placeholder = chrome.i18n.getMessage('Create__a__password'),
+  autoFocus = false,
+  className,
```
className should be removed as we've migrated all of that out of the extension. That's old MUI

tombeckenham commented on src/ui/views/Setting/change-password/index.tsx:
```diff
@@ -291,198 +320,50 @@ const ChangePassword = () => {
-                '&.Mui-focused': {
-                  border: '1px solid #FAFAFA',
-                  boxShadow: '0px 8px 12px 4px rgba(76, 76, 76, 0.24)',
+          <PasswordForm
```
I don't like the way this was done... It's not very React like. Could you get rid of PasswordForm and just include the PasswordInput components directly?

Commit:
fixed: remove extra sx

Closes #861

Commit:
feat: update password with helper text and error text component

Closes #861

tombeckenham committed:
Added helper text and error text

Closes #861

tombeckenham committed:
Updated unlock to use errorText

Closes #861

tombeckenham committed:
refactor: update PasswordInput props to use showPassword and setShowPassword

This change standardizes the prop names for password visibility across multiple components, enhancing consistency and readability.

Closes #861

tombeckenham committed:
Updated remove profile dialog to use errorText

Closes #861

tombeckenham committed:
SettingsPassword uses errorText

Closes #861

tombeckenham committed:
Updated change password dialog

Closes #861

tombeckenham committed:
Updated DecryptWallet

Closes #861

tombeckenham committed:
Corrected error handling logic in setpassword

Closes #861

tombeckenham committed:
Corrected infinite loop in change password

Closes #861

tombeckenham committed:
Removed redundant components

Closes #861

tombeckenham committed:
Clear opossing message states

Closes #861

tombeckenham committed:
Corrected error message naming and dependecies in changePasswordWithBackups

Closes #861

tombeckenham committed:
Can't have rows if minRows is defined

Closes #861

tombeckenham committed:
Replaced SlideRelative with Collapse and reverted SlideRelative for other components

Closes #861

tombeckenham committed:
Added password component stories

Closes #861

tombeckenham committed:
Updated imports of SettingsPassword

Closes #861

tombeckenham committed:
Refactored password validation logic in SetPassword component to handle empty inputs and improve error messaging

Closes #861

tombeckenham committed:
Enhanced layout of ChangePassword component by adding gap styling and refining placeholder text for better clarity

Closes #861

tombeckenham committed:
Updated SetPassword component and added story

Closes #861

tombeckenham committed:
Updated gap and removed formgroup

Closes #861

tombeckenham committed:
Now checks if touched before showing validation error and helper text

Closes #861

tombeckenham committed:
Added readonly story

Closes #861

tombeckenham committed:
Click agree to terms

Closes #861

tombeckenham committed:
Refactored checkbox interactions in helper functions to use setChecked for better clarity and consistency

Closes #861

tombeckenham committed:
Added the password label back into e2e tests

Closes #861

tombeckenham committed:
Reverted change to show errors only on blur

Closes #861
